app/helpers/azure_compute_resource_helper.rb

```diff
@@ -0,0 +1,13 @@
+module AzureComputeResourceHelper
+  def regions_list(azurerm_cr, f)
+    begin
+      regions = azurerm_cr.regions || []
+    rescue AuthenticationError
+      #do nothing
+      regions = []
+    rescue Exception => ex
+      return information_box("Regions could not be loaded due to exception: #{ex}")
+    end
+    selectable_f(f, :url, regions, {}, {:label => _('Azure Region'), :disabled => regions.empty?, :required => true, :help_inline_permanent => load_button_f(f, regions.present?, _("Load Regions")) })
```

Comment:
Made changes as per suggestions. Although, the[test_connection_validation](https://github.com/theforeman/foreman/blob/develop/app/models/compute_resource.rb#L112) returns `false` since region is not present for the first time and hence, causes `can't be blank` to appear. 
This behaviour has also been observed in GCE.